src/ledger/LedgerManagerImpl.cpp

```diff
@@ -858,6 +872,45 @@ LedgerManagerImpl::setupLedgerCloseMetaStream()
         }
     }
 }
+void
+LedgerManagerImpl::maybeResetLedgerCloseMetaDebugStream(uint32_t ledgerSeq)
+{
+    if (mApp.getConfig().METADATA_DEBUG_LEDGERS != 0)
+    {
+        if (mMetaDebugStream)
+        {
+            if (!FlushAndRotateMetaDebugWork::isDebugSegmentBoundary(ledgerSeq))
+            {
+                // If we've got a stream open and aren't at a reset boundary,
+                // just return -- keep streaming into it.
+                return;
+            }
+
+            // If we are resetting and already have a stream, hand it off to the
+            // flush-and-rotate work to finish up with.
+            mApp.getWorkScheduler().scheduleWork<FlushAndRotateMetaDebugWork>(
```

Comment:
Hm. So we would only write to the stream if there's no close-and-rotate running? Possibly. This would almost never produce gaps during normal operation (you have 5 seconds to finish each close-and-rotate, will almost always succeed) but it would very often produce streams with gaps in them during catchup.

An alternative would be to skip a close-and-rotate launch if one is already running and wait for the next segment boundary to try again -- keep the stream open and keep writing to it. This would cause (probably only slight) size increases in each meta file during catchup -- I'd imagine only a factor of 2 or 3 at worst -- and would produce streams with no gaps. Given we're only talking about maybe 5GB of disk for this when it's turned on on big production nodes, I think causing 10 or 15GB during catchup is fairly harmless. But I could go either way on it, whichever you prefer.

Reply:
(I implemented this latter variant, but if you want me to change it to the one you suggested, say so!)